test/jdk/java/foreign/TestBufferStackStress2.java

```diff
@@ -62,6 +62,16 @@ final class TestBufferStackStress2 {
      */
     @Test
     void movingVirtualThreadWithGc() throws InterruptedException {
+
+        // If the VM is configured such that the main thread is virtual,
+        // this stress test will not work as the main thread is always alive causing
+        // us to wait forever for contraction.
+        // Hence, we will skipp this test if the main thread is virtual.
+        if (Thread.currentThread().isVirtual()) {
+            System.out.println("Skipped because the main thread is a virtual thread");
+            return;
+        }
```

Comment:
This is okay. An alternative that we use in a few other tests is to use the Assumptions API t, e.g. assumeFalse(Thread.currentThread().isVirtual(), "message").